File: src/dataset/javascript/series_exercism-ThomasZumsteg/openAI/ChatGPT4_series_v4.js
class Series {
    constructor(digits) {
        this.digits = Array.from(digits, Number);
    }

    slices(sliceSize) {
        if (sliceSize > this.digits.length) {
            throw new Error("Slice size is too big.");
        }
        const result = [];
        for (let i = 0; i <= this.digits.length - sliceSize; i++) {
            result.push(this.digits.slice(i, i + sliceSize));
        }
        return result;
    }
}

module.exports = Series;

File: src/dataset/python/series_exercism-ThomasZumsteg/openAI/ChatGPT4_series_v4.py
"""Finds consecutive number sequences"""

def slices(digits, size):
    """Returns list of lists of consecutive digits"""
    if size <= 0 or size > len(digits):
        raise ValueError

    return [list(map(int, digits[i:i+size])) for i in range(len(digits) - size + 1)]